### PR TITLE
SER-3453 Track addtional data to discern crosswiki search

### PR DESCRIPTION
**Pull request by mateuszsip**
## Links

* https://wikia-inc.atlassian.net/browse/SER-3453
* https://wikia-inc.atlassian.net/browse/DE-4494

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
   "dependencies": {
     "@wikia/ad-engine": "git+https://github.com/Wikia/ad-engine.git#v41.9.3",
     "@wikia/ember-fandom": "github:Wikia/ember-fandom#20adaf9b214b67c1046090b6caed91aec92c472a",
-    "@wikia/search-tracking": "github:Wikia/search-tracking#1.2.0",
```
Comment (mateuszsip) [inline package.json:38]: Use version tag after https://github.com/Wikia/search-tracking/pull/10 is released

File: app/controllers/search.js [hunk 2]
```diff
@@ -43,15 +43,17 @@ export default Controller.extend({
       searchPhrase: this.inputPhrase,
       clicked: {
         type: 'article', // currently the only displayed type in the search
-        id: result.id,
+        id: result.wikiId + '_' + result.id,
```
Comment (slayful) [inline app/controllers/search.js]: Won't this break existing tracking?

File: app/controllers/search.js [hunk 3]
```diff
@@ -43,15 +43,17 @@ export default Controller.extend({
...
         thumbnail: false, // we do not show thumbnails on SRP right now
-        wikiId: result.wikiId,
+      },
+      filters: {
+          searchType: this.model.getScope()
       },
       target: 'redirect',
       app: 'mw-mobile',
-      siteId: result.wikiId,
+      siteId: this.wikiVariables.id,
```
Comment (slayful) [inline app/controllers/search.js:56]: So we pass the ID of the wiki we're currently on?

File: app/controllers/search.js (hunk 2 above)
Comment (slayful) [reply app/controllers/search.js]: `result` is the object from the model or from the view?

File: app/controllers/search.js [hunk 4]
```diff
@@ -43,15 +43,17 @@ export default Controller.extend({
       searchPhrase: this.inputPhrase,
       clicked: {
         type: 'article', // currently the only displayed type in the search
-        id: result.id,
+        id: result.wikiId + '_' + result.id,
         title: result.title,
         position: result.position + 1, // +1 since we need to start with 1 instead of 0
         thumbnail: false, // we do not show thumbnails on SRP right now
```
Comment (slayful) [inline app/controllers/search.js:49]: I don't see anything that would indicate that it's a cross-wiki article search other than the ID including the wiki ID of possibly another wiki. If that's the only indication how do we differentiate clicks on articles from wiki the user is currently on in both search modes?

File: app/controllers/search.js [hunk 5]
```diff
@@ -69,9 +71,11 @@ export default Controller.extend({
     const batchEnd = batchBegin + batchSize;
     const payload = {
       searchPhrase: this.inputPhrase,
-      filters: {}, // there is no way in mobile-wiki to set any filter
+      filters: {
+        searchType: this.model.getScope()
+      }, // there is no way in mobile-wiki to set any filter
```
Comment (slayful) [inline app/controllers/search.js]: Well now there is, right? :P

**Review by slayful — CHANGES_REQUESTED** [review]
I have some concerns about differentiating between cross-wiki and on-wiki article search

File: app/controllers/search.js (hunk 2 above)
Comment (mateuszsip) [reply app/controllers/search.js]: https://wikia-inc.atlassian.net/browse/DE-4494 - that's the blocker I mentioned on daily - we agreed on this contract during meeting with DE last week.

File: app/controllers/search.js (hunk 3 above)
Comment (mateuszsip) [reply app/controllers/search.js:56]: Yes, and `clicked.id` is a composite key combining wiki id and page id.

**Review by slayful — APPROVED** [review]
My concerns have been addressed, the filter is included in the payload and the data will be migrated.

File: app/controllers/search.js (hunk 5 above)
Comment (mateuszsip) [inline app/controllers/search.js]:
```suggestion
      },
```

**Review by rybmat — APPROVED** [review]

**Pull request merged**